pkg/client/actionclient.go

```diff
@@ -61,13 +61,71 @@ type GetOption func(*action.Get) error
 type InstallOption func(*action.Install) error
 type UpgradeOption func(*action.Upgrade) error
 type UninstallOption func(*action.Uninstall) error
+type RollbackOption func(*action.Rollback) error
 
-func NewActionClientGetter(acg ActionConfigGetter) ActionClientGetter {
-	return &actionClientGetter{acg}
+type ActionClientGetterOption func(*actionClientGetter) error
+
+func AppendGetOptions(opts ...GetOption) ActionClientGetterOption {
+	return func(getter *actionClientGetter) error {
+		getter.defaultGetOpts = append(getter.defaultGetOpts, opts...)
+		return nil
+	}
+}
+
+func AppendInstallOptions(opts ...InstallOption) ActionClientGetterOption {
+	return func(getter *actionClientGetter) error {
+		getter.defaultInstallOpts = append(getter.defaultInstallOpts, opts...)
+		return nil
+	}
+}
+
+func AppendUpgradeOptions(opts ...UpgradeOption) ActionClientGetterOption {
+	return func(getter *actionClientGetter) error {
+		getter.defaultUpgradeOpts = append(getter.defaultUpgradeOpts, opts...)
+		return nil
+	}
+}
+
+func AppendUninstallOptions(opts ...UninstallOption) ActionClientGetterOption {
+	return func(getter *actionClientGetter) error {
+		getter.defaultUninstallOpts = append(getter.defaultUninstallOpts, opts...)
+		return nil
+	}
+}
+
+func AppendInstallFailureUninstallOptions(opts ...UninstallOption) ActionClientGetterOption {
+	return func(getter *actionClientGetter) error {
+		getter.installFailureUninstallOpts = append(getter.installFailureUninstallOpts, opts...)
+		return nil
+	}
+}
+func AppendUpgradeFailureRollbackOptions(opts ...RollbackOption) ActionClientGetterOption {
+	return func(getter *actionClientGetter) error {
+		getter.upgradeFailureRollbackOpts = append(getter.upgradeFailureRollbackOpts, opts...)
+		return nil
+	}
+}
+
+func NewActionClientGetter(acg ActionConfigGetter, opts ...ActionClientGetterOption) (ActionClientGetter, error) {
+	actionClientGetter := &actionClientGetter{acg: acg}
+	for _, opt := range opts {
+		if err := opt(actionClientGetter); err != nil {
+			return nil, err
+		}
+	}
+	return actionClientGetter, nil
 }
 
 type actionClientGetter struct {
 	acg ActionConfigGetter
+
+	defaultGetOpts       []GetOption
+	defaultInstallOpts   []InstallOption
+	defaultUpgradeOpts   []UpgradeOption
+	defaultUninstallOpts []UninstallOption
+
+	installFailureUninstallOpts []UninstallOption
+	upgradeFailureRollbackOpts  []RollbackOption
```

Comment:
This PR looks great overall. The only thing that keeps me wondering is that I suspect the lack/presence of the `default` prefix for these fields is trying to convey something, but I'm not sure what. Perhaps I'm just not familiar enough with the code...